sdk/search/Azure.Search.Documents/tests/Samples/Sample05_IndexingDocuments.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.Core.Tests.TestFramework;
+using Azure.Search.Documents.Indexes;
+using Azure.Search.Documents.Indexes.Models;
+using NUnit.Framework;
+
+#pragma warning disable SA1402 // File may only contain a single type
+#pragma warning disable SA1649 // File name should match first type name
+
+namespace Azure.Search.Documents.Tests.Samples
+{
+    #region Snippet:Azure_Search_Documents_Tests_Samples_Sample05_IndexingDocuments_LegacyProduct
+    public class Product
+    {
+        [SearchableField(IsKey = true)]
+        public string Id { get; set; }
+
+        [SearchableField(IsFilterable = true)]
+        public string Name { get; set; }
+
+        [SimpleField(IsSortable = true)]
+        public double Price { get; set; }
+
+        public override string ToString() =>
+            $"{Id}: {Name} for {Price:C}";
+    }
+    #endregion
+
+    public class Sample05_IndexingDocuments : SearchTestBase
+    {
+        public Sample05_IndexingDocuments(bool async, SearchClientOptions.ServiceVersion serviceVersion)
+            : base(async, serviceVersion, null /* RecordedTestMode.Record /* to re-record */)
+        {
+        }
+
+        #region Snippet:Azure_Search_Documents_Tests_Samples_Sample05_IndexingDocuments_GenerateCatalog
+        public IEnumerable<Product> GenerateCatalog(int count = 1000)
+        {
+            // Adapted from https://weblogs.asp.net/dfindley/Microsoft-Product-Name-Generator
```

Comment:
I feel like we should put this in Azure.Core.TestFramework. 😁

Reply:
That's a good idea for a future release - a lot of the internal test frameworks I've used had really good random generators for names, addresses, etc., that made it easy to set things like this up out of the box.